scripts/build-ancillary-repository.sh

```diff
@@ -142,10 +142,28 @@ function build_ancillary_repository() {
 #    environment variables, and the script will work as expected.
 #
 
-if [[ -z "$UPSTREAM_BRANCH" ]]; then
-	echo "UPSTREAM_BRANCH is not set."
-	exit 1
+#
+# Set UPSTREAM_BRANCH. This will determine which version of the linux package
+# mirror is used.
+#
+if [[ -z "$UPSTREAM_PRODUCT_BRANCH" ]]; then
+	echo "UPSTREAM_PRODUCT_BRANCH is not set."
+	if ! source "$TOP/branch.config" 2>/dev/null; then
+		echo "No branch.config file found in repo root."
+		exit 1
+	fi
+
+	if [[ -z "$UPSTREAM_BRANCH" ]]; then
+		echo "UPSTREAM_BRANCH parameter was not sourced from branch.config." \
+			"Ensure branch.config is properly formatted with e.g." \
+			"UPSTREAM_BRANCH=\"<upstream-product-branch>\""
+		exit 1
+	fi
+	echo "Defaulting to branch $UPSTREAM_BRANCH set in branch.config."
+else
+	UPSTREAM_BRANCH="$UPSTREAM_PRODUCT_BRANCH"
```

Comment:
It might be cleaner we could add this logic to the "gradlew" wrapper. This way it will be set for all gradle Tasks and we do not need to worry about inconsistencies if other tasks/scripts end up needing this environment variable.

Reply:
We should not be modifying `gradlew`.

IMO, it would have been best to have had this logic added to the gradle build logic to begin with, but for whatever reason it was added to `docker-run.sh` (which we're no longer using). If you'd like to open a separate PR to move this into the gradle build logic, then I can rebase on to that PR, and remove this.. Otherwise, I'd like to move forward with what I have here.

Reply:
The reason this was added to `docker-run.sh` instead of the gradle logic is because there was no straight forward way to add logic in gradle to set environment variables for all sub-tasks. 

Do you think adding a wrapper on top of gradlew that would be responsible of setting the Environment the way we need would be a viable option?

Reply:
> Do you think adding a wrapper on top of gradlew that would be responsible of setting the Environment the way we need would be a viable option?

To me, that seems awfully complicated and awkward.. given that we're already leveraging a complicated and powerful build system (Gradle).

If I understand you correctly, we'd call into a shell script, which would then call into Gradle, which would then call into another shell script. This just seems unnecessarily complicated and convoluted to me.

But if you want to do the work to change this, then, by all means...

Reply:
Given that our build system was designed to depend on configuration to be passed through the environment, this wrapper script would just make sure that the configuration is sane and fill-in defaults (such as with the UPSTREAM_BRANCH parameter), so while it would add one more layer of complexity, the layer is fairly thin and simple.

That said, given that we don't have a consensus on this, I'm going to drop this issue for now so that we can proceed with this change.

Reply:
Thanks.

If we have time, maybe we can discuss this at our next team meeting.. or set up another meeting with @jgallag88 (and anybody else that's interested) to come up with a more maintainable solution than what I'm doing here.

Reply:
We have done something like what Pavel is suggesting in the past (see `appliance/java-home-helper.diff` and `appliance/gradle/java-home-helper.sh` in the app-gate). It's not the end of the world, but it is a bit of a burden to maintain because we have to keep applying our diff to the wrapper every time we upgrade Gradle.

I think the simplest thing to do would be to move the logic to choose the correct value into a gradle function, and then use that to set the environment variable in the appropriate `Exec` tasks.

That said, I'm on board with pushing this as is. We can do a follow up if we want to.